libraries/provider_hab_service.rb

```diff
@@ -0,0 +1,69 @@
+# Copyright:: Copyright 2016, Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/provider/service"
+
+class Chef
+  class Provider
+    class Service
+      class HabSup < Chef::Provider::Service
+        include HabitatCookbook::Helpers
+
+        use_inline_resources
+
+        provides :hab_service
+
+        action :start do
+          declare_resource(:systemd_unit, "#{service_name}.service") do
+            action :create
+            content lazy { unit_content }
+            notifies :restart, "service[#{service_name}]"
+          end
+
+          declare_resource(:service, service_name) do
+            action :start
+          end
+        end
+
+        action :stop do
+          declare_resource(:service, service_name) do
+            action :stop
+          end
+        end
+
+        def service_name
+          new_resource.service_name.split("/")[1]
+        end
+
+        def unit_content
+          return new_resource.content if new_resource.content
```

Comment:
you should probably rename `new_resource.content` here to `new_resource.unit_content`.  here you're delegating with fall through if its not set, so its appropriate to be named the same thing.  i'd also prefer the longer name since `content` is not specific enough and collides conceptually with the file resource `content` setting in my head (even though `systemd_unit` also uses `content`, but i think its probably a good idea to avoid overly generic property names in subclasses of core resources like service in case the superclass winds up wanting that name).